src/Bicep.Types/TypeLoader.cs

```diff
@@ -34,6 +34,16 @@ public ResourceFunctionType LoadResourceFunctionType(CrossFileTypeReference refe
             return resourceFunctionType;
         }
 
+        public ObjectType LoadObjectType(CrossFileTypeReference reference)
```

Comment:
Rather than adding a new method here, could you make `LoadType` public, and add it to the `ITypeLoader` interface?